### ISSUE TITLE
would you accept a PR to add async/await support?

### ISSUE THREAD

**Issue by mreinstein** [mcwhittemore/staged-git-files#19]
My goal: 

```javascript
async function main () {
   const stagedFiles = await sgf();
   // do stuff here with them files...
}

main();
```

Here's what I'd like to do to the API:

```javascript
sgf(filter, callbackFn);  // callback usage
sgf(callbackFn);          // callback usage

sgf(filter);   // returns a promise
sgf();         // returns a promise
```

Would this be a welcome addition? Happy to submit a PR.

**Comment by mcwhittemore** [comment]
Yea, this would be great!

**Comment by mreinstein** [comment]
resolved via #20
